src/gui/transferlistfilterswidget.cpp

```diff
@@ -148,6 +148,8 @@ StatusFilterWidget::StatusFilterWidget(QWidget *parent, TransferListWidget *tran
 {
     connect(BitTorrent::Session::instance(), &BitTorrent::Session::torrentsUpdated
             , this, &StatusFilterWidget::updateTorrentNumbers);
+    connect(BitTorrent::Session::instance(), &BitTorrent::Session::torrentAboutToBeRemoved
```

Comment:
What about `BitTorrent::Session::torrentAdded()` signal?

Reply:
From what I see, the `torrentsUpdated` will be emitted at least once (even for paused torrent).

Reply:
I would rather have a reliable code path than depend on some assumptions.

Reply:
>I would rather have a reliable code path than depend on some assumptions.

I think the behavior I observed is implied by libtorrent doc under `state_update_alert`:
>It contains the torrent status of all torrents that changed since last time this message was posted. 

The status changed from `<nothing>` to `paused/running state`.

Reply:
>I think the behavior I observed is implied by libtorrent doc

The main idea is the components outside libtorrent abstraction layer should know nothing about libtorrent. Otherwise we miss all advantages.

Besides, I really don't understand why you're so stubborn... Why don't you want to implement it explicitly? Sorry, I insist on it.
We already have no enough resources for maintaining the code to plant time bombs on ourselves.

Reply:
>Besides, I really don't understand why you're so stubborn... Why don't you want to implement it explicitly?

You never give a compelling argument until your last post. Or do you advise everyone just drop their thoughts and listen to you?

>The main idea is the components outside libtorrent abstraction layer should know nothing about libtorrent.

This reason is justified. PR updated.

Reply:
>You never give a compelling argument until your last post.

Sorry, I thought it was pretty clear that the main interaction of components should be based on their interfaces. To hope for any concrete details of implementation is an unreliable way. And there have already been examples of the unfortunate outcome of such cases (more recently, when some people based the logic of their web client on the hardcoded lifetime of qBittorrent web session).

>we don't have clear definition on signal behaviors, different people have different thoughts of it.

Then we should discuss it.
>And even looking at source code is not enough to describe it...

Ideally you should never look at the source code of some component just to use it.

Reply:
>>we don't have clear definition on signal behaviors, different people have different thoughts of it.
>
>Then we should discuss it.

On a second thought, maybe there is no need for now. I overlooked that the torrent update signal interval can be configured to a long value and if I rely on it then the numbers in UI would be incorrect for a long time. So far, there shouldn't be any different views on the expected signal behavior.